### ISSUE TITLE
5.0+ versions do not restore realtime mode when stopping to navigate via SpaceMouse

### ISSUE THREAD

**Issue by aknittel1** [microdee/UE4-SpaceMouse#72]
Currently, if I set RealTime off in the Viewport and then touch my Spacemouse to navigate, it immediately turns it ON and stays ON. 

I reported this in discussions.

**Comment by aknittel1** [comment]
Is it possible to get a general answer on this? I know there was some debate as to implementing this in the other thread, and I'm only assuming that the decision was to honor the REALTIME user setting and return it to the previous state after navigating. Is this how it's also supposed to behave with the versions supporting 5.0+ ?

**Comment by microdee** [comment]
I have a bad news mate, Realtime Off restores for me once I stop movement as it was also doing prior 5.0. Can you tell me what spacemouse device model are you using?

Also not restoring might indicate slight drift of the puck, you can re-calibrate your device if you're pressing the "Menu Options" or "Left" button and the "View Fit" or "Right" button at the same time and hold until the blue light under the puck is blinking.

**Comment by aknittel1** [comment]
Spacemouse wireless. Just recalibrated as you suggested--same issue. I haven't updated the driver from 3Dconnection in a long while. Are you dependent on on their driver or do you bypass?

**Comment by aknittel1** [comment]
Just upgraded - no difference. 
![image](https://user-images.githubusercontent.com/22087392/158684447-fc9b9aac-2c27-4805-a178-23d4c84c0e0e.png)

**Comment by aknittel1** [comment]
Here is some debug info. I had turned realtime off, then moved it.

![image](https://user-images.githubusercontent.com/22087392/158685136-5d3c592d-4d79-4e5d-859d-5f519708aaab.png)

**Comment by microdee** [comment]
> Spacemouse wireless. Just recalibrated as you suggested--same issue. I haven't updated the driver from 3Dconnection in a long while. Are you dependent on on their driver or do you bypass?

I don't use their SDK or services, I read data off those devices with USB/HID. I'll only have time to look at what might go on here more thoroughly during the weekend. As of now I have no idea why it doesn't switch back to non-realtime in your case. You mentioned only 5.0, did that work before in 4.xx for you?

**Comment by aknittel1** [comment]
That's fine, I appreciate it. I never had 4+ installed. I started with UNREAL at 5EA. I have 5 preview 2 installed now and focusing on that.

**Comment by aknittel1** [comment]
Here's my DefaultEditor.ini file in case you need to review my settings:

[/Script/SpaceMouse.SpaceMouseConfig]
CameraBehavior=OrbitingNoRoll
MovementSecondsTolerance=0.010000
RotationDegreesPerSec=20.000000
PitchAxisMap=(X=0.000000,Y=1.000000,Z=0.000000)
RollAxisMap=(X=0.000000,Y=0.000000,Z=0.000000)
TranslationUnitsPerSec=500.000000
XTranslationAxisMap=(X=0.000000,Y=1.000000,Z=0.000000)
YTranslationAxisMap=(X=-1.000000,Y=0.000000,Z=0.000000)
ZTranslationAxisMap=(X=0.000000,Y=0.000000,Z=1.000000)
YawAxisMap=(X=0.000000,Y=0.000000,Z=-1.000000)
MaxHidReadOperationsPerFrame=2048
DisplayDebugInformation=False
ActiveInBackground=False

**Comment by microdee** [comment]
set `MovementSecondsTolerance=1` and see what happens

**Comment by aknittel1** [comment]
That fixed it. 
You're probably wondering why I set that so low. I had downloaded the Quixel Megascan Medieval Forest environment and wondered why navigation was jerky, so I was experimenting with that setting but to no avail. I moved over to a not so intensive environment and it was smooth, so I don't need a setting that I don't quite understand and didn't help the jerky movement. I have a very high end rig but I understand  that some environments with everything animated (wind through vegetation, smole, fire) can make even my rig not service requests from HID devices smoothly. ANyway I don't need Realtime on, i'm a filmmaker not a gamer or game developer. Stuff will animate when I render.
